autonav_ws/src/autonav_hardware/src/can_node.py

```diff
@@ -0,0 +1,289 @@
+#!/usr/bin/env python3
+
+import rclpy
+from autonav_shared.node import Node
+from autonav_msgs.msg import MotorInput, MotorFeedback, SafetyLights, Conbus
+from autonav_msgs.msg import LinearPIDStatistics, AngularPIDStatistics, MotorStatisticsFrontMotors, MotorStatisticsBackMotors
+from autonav_shared.types import LogLevel, DeviceState, SystemState
+import can
+import threading
+import struct
+from ctypes import Structure, c_bool, c_uint8
+
+arbitration_ids = {
+    "EStop": 0,
+    "MobilityStop": 1,
+    "MobilityStart": 9,
+    "MotorsCommand": 10,
+    "SafetyLightsCommand": 13,
+    "OdometryFeedback": 14,
+    "ObjectDetection": 20,
+    "LinearPIDStatistics": 50,
+    "AngularPIDStatistics": 51,
+    "MotorStatisticsFrontMotors": 52,
+    "MotorStatisticsBackMotors": 53,
+    "ConbusLowerBound": 1000,
+    "ConbusUpperBound": 1400
+}
+
+class SafetyLightsPacket(Structure):
+    _fields_ = [
+        ("autonomous", c_bool, 1),
+        ("mode", c_uint8, 7),
+        ("brightness", c_uint8, 8),
+        ("red", c_uint8, 8),
+        ("green", c_uint8, 8),
+        ("blue", c_uint8, 8),
+        ("blink_period", c_uint8, 8)
+    ]
+
+class CanConfig:
+    def __init__(self):
+        self.odom_feedback_scaler = 10000
+        self.linear_pid_scaling_factor = 1000
+        self.angular_pid_scaling_factor = 1000
+
+
+class can_node(Node):
+    def __init__(self):
+        super.__init__("CAN_node")
+        self.write_config(CanConfig)
+
+        # can
+        self.can = None
+
+        # safety lights
+        self.safetyLightsSubscriber = self.create_subscription(
+            SafetyLights,
+            "autonav/safety_lights",
+            self.on_safety_lights_received(),
```

Comment:
remove parentheses (and on following lines)  as this is calling the function instead of passing the function as a reference python thingamajig